pallets/subtensor/src/swap.rs

```diff
@@ -92,6 +82,81 @@ impl<T: Config> Pallet<T> {
         Ok(Some(weight).into())
     }
 
+    /// Swaps the coldkey associated with a set of hotkeys from an old coldkey to a new coldkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `origin` - The origin of the call, which must be signed by the old coldkey.
+    /// * `old_coldkey` - The account ID of the old coldkey.
+    /// * `new_coldkey` - The account ID of the new coldkey.
+    ///
+    /// # Returns
+    ///
+    /// Returns a `DispatchResultWithPostInfo` indicating success or failure, along with the weight consumed.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The caller is not the old coldkey.
+    /// - The new coldkey is the same as the old coldkey.
+    /// - The new coldkey is already associated with other hotkeys.
+    /// - The transaction rate limit for coldkey swaps has been exceeded.
+    /// - There's not enough balance to pay for the swap.
+    ///
+    /// # Events
+    ///
+    /// Emits a `ColdkeySwapped` event when successful.
+    ///
+    /// # Weight
+    ///
+    /// Weight is tracked and updated throughout the function execution.
+    pub fn do_swap_coldkey(
+        origin: T::RuntimeOrigin,
+        old_coldkey: &T::AccountId,
+        new_coldkey: &T::AccountId,
+    ) -> DispatchResultWithPostInfo {
+        ensure_signed(origin)?;
+
+        let mut weight = T::DbWeight::get().reads(2);
+
+        // Check if the new coldkey is already associated with any hotkeys
+        ensure!(
+            !Self::coldkey_has_associated_hotkeys(new_coldkey),
+            Error::<T>::ColdKeyAlreadyAssociated
+        );
+
+        let block: u64 = Self::get_current_block_as_u64();
+
+        // Swap coldkey references in storage maps
+        Self::swap_total_coldkey_stake(old_coldkey, new_coldkey, &mut weight);
+        Self::swap_stake_for_coldkey(old_coldkey, new_coldkey, &mut weight);
+        Self::swap_owner_for_coldkey(old_coldkey, new_coldkey, &mut weight);
+        Self::swap_total_hotkey_coldkey_stakes_this_interval_for_coldkey(
```

Comment:
Maybe add comment about order of these swaps being important

Reply:
done